javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/Reference.java

```diff
@@ -363,25 +364,36 @@ public boolean matches(int asmFlags) {
     NON_FINAL {
       @Override
       public boolean contradicts(Flag anotherFlag) {
-        return anotherFlag == FINAL;
+        return anotherFlag == FINAL || anotherFlag == ABSTRACT;
       }
 
       @Override
       public boolean matches(int asmFlags) {
-        return (Opcodes.ACC_FINAL & asmFlags) == 0;
+        return ((Opcodes.ACC_ABSTRACT | Opcodes.ACC_FINAL) & asmFlags) == 0;
       }
     },
     FINAL {
       @Override
       public boolean contradicts(Flag anotherFlag) {
-        return anotherFlag == NON_FINAL;
+        return anotherFlag == NON_FINAL || anotherFlag == ABSTRACT;
       }
 
       @Override
       public boolean matches(int asmFlags) {
         return (Opcodes.ACC_FINAL & asmFlags) != 0;
       }
     },
+    ABSTRACT {
```

Comment:
Can you add javadoc to methods of `Flag` enum?

Reply:
I'll add javadoc to `matches()` method -- I've checked the usages of all of them and it turned out that `contradicts()` and `supersedes()` are never used. I guess that I'll be doing a general code clean-up together with muzzle documentation 😄